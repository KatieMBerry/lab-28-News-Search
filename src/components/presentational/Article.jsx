import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description, urlToImage }) => (
    <article>
        <h3 style={{ color: '#c13e72' }}>{title}</h3>
        <img src={urlToImage} style={{ width: '400px', height: '300px' }} />
        <h4 style={{ color: '#1ca69d' }}>{author}</h4>
        <p>{description}</p>
    </article>
);

Article.propTypes = {
    title: PropTypes.string.isRequired,
    urlToImage: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
};

export default Article;
