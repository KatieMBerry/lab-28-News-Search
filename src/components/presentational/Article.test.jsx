import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
    afterEach(() => cleanup());
    it('renders Article', () => {
        const { asFragment } = render(<Article
            title='New Title'
            author='WSJ'
            description='Inquiring minds...'
            urlToImage='urlToMyImage'
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});
