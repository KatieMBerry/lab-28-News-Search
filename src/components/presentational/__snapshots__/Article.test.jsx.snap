// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <article>
    <h3
      style="color: rgb(193, 62, 114);"
    >
      New Title
    </h3>
    <img
      src="urlToMyImage"
      style="width: 400px; height: 300px;"
    />
    <h4
      style="color: rgb(28, 166, 157);"
    >
      WSJ
    </h4>
    <p>
      Inquiring minds...
    </p>
  </article>
</DocumentFragment>
`;
